src/frontend/src/app/registration/registration.component.html

```diff
@@ -0,0 +1,68 @@
+      <div *ngIf="registrationForm.controls.firstname.invalid && (registrationForm.controls.firstname.touched || registrationForm.controls.firstname.dirty)">
+      <small id="firstnameError" class="form-text text-danger" *ngIf="registrationForm.controls.firstname.errors.required">firstname is required.</small>
+      </div>
+      </div>
+
+    <div class="form-group">
+      <label for="inputLastname">Lastname</label>
+      <input type="text" class="form-control w-50" id="inputLastname"
+             placeholder="Enter lastname" [(ngModel)]="userDto.lastname" formControlName="lastname">
+      <div *ngIf="registrationForm.controls.lastname.invalid && (registrationForm.controls.lastname.touched || registrationForm.controls.lastname.dirty)">
+      <small id="lastnameError" class="form-text text-danger" *ngIf="registrationForm.controls.lastname.errors.required">lastname is required.</small>
+      </div>
+    </div>
+      <div class="form-group">
+      <label for="inputUsername">Username</label>
+      <input type="text" class="form-control w-50" id="inputUsername" aria-describedby="usernameHelp"
+             placeholder="Enter username" [(ngModel)]="userDto.username" formControlName="username">
+      <small id="usernameHelp" class="form-text text-muted">provide username that will be used to login</small>
+      <div *ngIf="registrationForm.controls.username.invalid && (registrationForm.controls.username.touched || registrationForm.controls.username.dirty)">
+      <small id="usernameError" class="form-text text-danger" *ngIf="registrationForm.controls.username.errors.required">username is required.</small>
+      <small id="usernameError" class="form-text text-danger" *ngIf="registrationForm.controls.username.errors.minlength || registrationForm.controls.username.errors.maxlength">username must be between 5 to 9 characters.</small>
+      </div>
+    </div>
+
+    <!--<div class="form-group">-->
+      <!--<label for="inputEmail">Email</label>-->
+      <!--<input type="email" class="form-control w-50" id="inputEmail" aria-describedby="emailHelp"-->
+             <!--placeholder="Enter email" [(ngModel)]="userDto.email" name="email" required>-->
+      <!--<small id="emailHelp" class="form-text text-muted">provide email to sent you verification code.</small>-->
+    <!--</div>-->
+
+    <div class="form-group">
+      <label for="inputPassword">Password</label>
+      <input type="password" class="form-control w-50" id="inputPassword" aria-describedby="passwordHelp"
+             [(ngModel)]="userDto.password" formControlName="password">
+      <small id="passwordHelp" class="form-text text-muted">Password must consist of at least one big character, one number and has to be 8 character's long.</small>
+      <div *ngIf="registrationForm.controls.password.invalid && (registrationForm.controls.password.touched || registrationForm.controls.password.dirty)">
```

Comment:
ten if musi się znajdować w kontrolerze czyli w komponencie, stwórz na to jakąś funkcję, a tu tylko ngIf=i nazwa funkcji